### ISSUE TITLE
Detect and eliminate blank columns after the last field 

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/FreeCENMigration#1238]
Regardless  I can see that there were a number of uploads of the file before validation commenced and it was during those that the additional columns were added to the file

The initial upload had the headers as expected and the file commenced with the 3rd ED.
enumeration_district,civil_parish,ecclesiastical_parish,where_census_taken,ward,parliamentary_constituency,sanitary_district,location_flag,folio_number,page_number,schedule_number,uninhabited_flag,house_number,house_or_street_name,rooms,address_flag,surname,forenames,name_flag,relationship,marital_status,sex,age,individual_flag,occupation,occupation_category,occupation_flag,verbatim_birth_county,verbatim_birth_place,nationality,birth_county,birth_place,birth_place_flag,disability,language,notes
3,Brandesburton,-,Brandesburton,-,-,-,,21,1,1,,,The Hall,,,TONG,George,,Head,M,M,60,,Gardener,E,,YKS,Ellerby,,,,,,,

The next upload had additional blank columns in the deaders and the file now commenced with the 1st ED
enumeration_district,civil_parish,ecclesiastical_parish,where_census_taken,ward,parliamentary_constituency,sanitary_district,location_flag,folio_number,page_number,schedule_number,uninhabited_flag,house_number,house_or_street_name,rooms,address_flag,surname,forenames,name_flag,relationship,marital_status,sex,age,individual_flag,occupation,occupation_category,occupation_flag,verbatim_birth_county,verbatim_birth_place,nationality,birth_county,birth_place,birth_place_flag,disability,language,notes,,,,
1,Bonwick,All saints beeford,bonwick,,holderness,skirlaugh,,4,1,1,,,,,,Topham,Moses,,Head,M,M,55,,Farmer,r,,Yks,Weaverthorpe,,,,,,,,,,,

The 17th line shows the rows all had the extra blank fields with the blanks after the notes
1,Dunnington,st leonards beeford,dunnington,,holderness,skirlaugh,,,2,1,,,,,,Robinson,Frederick J,,Head,M,M,47,,Farm Manager,e,,Yks,haslem,,,,,,,pob not located,,,,

They remained there through a number of subsequent uploads.

Now when the file starts validation we see something interesting.

The header has the validation column after the additional blank columns
enumeration_district,civil_parish,ecclesiastical_parish,where_census_taken,ward,parliamentary_constituency,sanitary_district,location_flag,folio_number,page_number,schedule_number,uninhabited_flag,house_number,house_or_street_name,rooms,address_flag,surname,forenames,name_flag,relationship,marital_status,sex,age,individual_flag,occupation,occupation_category,occupation_flag,verbatim_birth_county,verbatim_birth_place,nationality,birth_county,birth_place,birth_place_flag,disability,language,notes,,,,,record_valid

whereas the data has it after the notes field
1,Dunnington,St Leonards Beeford,Dunnington,-,Holderness,Skirlaugh,,4,2,1,,,,,,ROBINSON,Frederick J,,Head,M,M,47,,Farm Manager,E,,YKS,Haslem,,,,,,,pob not located,false

Bottom line People need to be careful not to introduce blanks columns

But even more importantly CSVProc should be detecting and eliminating blank columns after the notes field.

E3 U6 S7 C6: P22

**Comment by Captainkirkdawson** [comment]
Attempted a fix and deployed to test3
